scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedTaskExecutor.java

```diff
@@ -116,6 +118,12 @@ public TaskResultImpl execute(TaskContext context, PrintStream outputSink, Print
 
             if (process != null) {
                 process.destroy();
+                try {
+                    //wait for twice the time of the cleanup process
+                    process.waitFor((new CleanupTimeoutGetter()).getCleanupTimeSeconds() * 2, TimeUnit.SECONDS);
+                } catch (Exception e) {
+                    //TODO log when this exception happens
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)